tests/integration/test_charm.py

```diff
@@ -38,6 +40,38 @@ async def test_image_relation(app: Application, test_charm: Application):
     await model.wait_for_idle([app.name], wait_for_active=True, timeout=30 * 60)
 
 
+def image_created_from_dispatch(
+    image_base: str, app_name: str, connection: Connection, dispatch_time: datetime
+) -> bool:
+    """Return whether there is an image created after dispatch has been called.
+
+    Args:
+        image_base: The Ubuntu image base.
+        app_name: The The application name that built the image.
+        connection: The OpenStack connection instance.
+        dispatch_time: Time when the image build was dispatched.
+
+    Returns:
+        Whether there exists an image that has been created after dispatch time.
+    """
+    image_name = IMAGE_NAME_TMPL.format(
+        IMAGE_BASE=image_base, APP_NAME=app_name, ARCH=_get_supported_arch().value
+    )
+    images: list[Image] = connection.search_images(image_name)
```

Comment:
Out of curiosity: would it be possible to have no images?

Reply:
Yup! In the test, if the image is not found (it is unexpected), the test will fail.
If the image is not found in the code (on relation joined the image is not found for a particular cloud), it will print out a warning log and return. But that's still okay because the image relation changed hook will trigger an image build! :D